spec/models/import/transactions/actual_and_refund_csv_row_spec.rb

```diff
@@ -0,0 +1,583 @@
+
+        expect(row.activity).to be_nil
+        expect(row.errors["Activity RODA Identifier"][1]).to include("cannot be found")
+      end
+    end
+  end
+
+  describe "#roda_identifier" do
+    it "returns the identifier string" do
+      csv_row = double(CSV::Row)
+      allow(csv_row).to receive(:field).with("Activity RODA Identifier").and_return("RODA-ID")
+      row = described_class.new(csv_row)
+
+      expect(row.roda_identifier).to eql("RODA-ID")
+    end
+
+    context "when the RODA Identifier is blank" do
+      it "returns blank" do
+        csv_row = double(CSV::Row)
+        allow(csv_row).to receive(:field).with("Activity RODA Identifier").and_return(nil)
+        row = described_class.new(csv_row)
+
+        expect(row.roda_identifier).to eql(:blank)
+      end
+    end
+  end
+
+  describe "#actual_value" do
+    it "returns the value from the csv row" do
+      csv_row = double(CSV::Row)
+      allow(csv_row).to receive(:field).with("Actual Value").and_return("10000.50")
+      row = described_class.new(csv_row)
+
+      expect(row.actual_value).to eql BigDecimal("10000.50")
+    end
+
+    context "when the value is not a number" do
+      it "returns the value from the csv row" do
+        csv_row = double(CSV::Row)
+        allow(csv_row).to receive(:field).with("Actual Value").and_return("not a number")
+        row = described_class.new(csv_row)
+
+        expect(row.actual_value).to eql "not a number"
+      end
+    end
+
+    context "when there is no value" do
+      it "returns blank" do
+        csv_row = double(CSV::Row)
+        allow(csv_row).to receive(:field).with("Actual Value").and_return(nil)
+        row = described_class.new(csv_row)
+
+        expect(row.actual_value).to eql(:blank)
+      end
+    end
+  end
+
+  describe "#refund_value" do
+    it "returns the value from the csv row" do
+      csv_row = double(CSV::Row)
+      allow(csv_row).to receive(:field).with("Refund Value").and_return("12030.95")
+      row = described_class.new(csv_row)
+
+      expect(row.refund_value).to eql BigDecimal("12030.95")
+    end
+
+    context "when the value is not a number" do
+      it "returns the value from the csv row" do
+        csv_row = double(CSV::Row)
+        allow(csv_row).to receive(:field).with("Refund Value").and_return("not a number")
+        row = described_class.new(csv_row)
+
+        expect(row.refund_value).to eql "not a number"
+      end
+    end
+
+    context "when there is no value" do
+      it "returns blank" do
+        csv_row = double(CSV::Row)
+        allow(csv_row).to receive(:field).with("Refund Value").and_return(nil)
+        row = described_class.new(csv_row)
+
+        expect(row.refund_value).to eql(:blank)
+      end
+    end
+  end
+
+  describe "#comment" do
+    it "returns the value from the csv row" do
+      row = described_class.new(valid_csv_row)
+
+      expect(row.comment).to eql("This is a comment.")
+    end
+
+    context "when the comment is empty" do
```

Comment:
What happens if the comment is an empty string instead of nil?

Reply:
My understanding here is that `CSV::Row`, which is the expected passed in object returns nil for empty values, you would have to do something else to get an empty string?